.github/workflows/build.yaml

```diff
@@ -16,24 +16,22 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        include:
-          # Rolling (source)
-          - ROS_DISTRO: rolling
-            BUILD_TYPE: source
-          # Jazzy (binary)
-          - ROS_DISTRO: jazzy
-            BUILD_TYPE: binary
-          # Iron (binary)
-          - ROS_DISTRO: iron
-            BUILD_TYPE: binary
+        ROS_DISTRO: [rolling, jazzy, iron]
+        BUILD_TYPE: [source, binary]
+        exclude:
+          - BUILD_TYPE: ${{ github.event_name == 'schedule' && 'source' || '' }}
```

Comment:
**Suggestion:** The conditional expression for `BUILD_TYPE` in the `exclude` matrix configuration uses a complex inline conditional that might be error-prone and hard to read. Consider simplifying this logic or using a separate step to set environment variables based on conditions, which can then be used in the matrix. [maintainability, importance: 7]
```suggestion
          exclude:
            - BUILD_TYPE: ${{ github.event_name == 'schedule' ? 'source' : '' }}
```